### PR TITLE
Use the internal Firebase client to send password reset

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* Uses Seths internal FB client to send the password reset on user create
* Eliminates a fail-in-the-middle scenario

### CODE REVIEW

File: pkg/controller/user/importbatch.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ func (c *Controller) HandleImportBatch() http.Handler {
 				continue
 			} else if created {
 				newUsers = append(newUsers, &batchUser)
+				if err := c.fbInternal.SendPasswordResetEmail(ctx, user.Email); err != nil {
```
Comment (sethvargo) [inline pkg/controller/user/importbatch.go:70]: It'd be good to know what quotas, if any, exist for this

**Review by mikehelmick — APPROVED** [review]

File: pkg/controller/user/update.go [hunk 2]
```diff
@@ -103,6 +103,12 @@ func (c *Controller) HandleUpdate() http.Handler {
 	})
 }
 
+func (c *Controller) renderUpdate(ctx context.Context, w http.ResponseWriter, user *database.User) {
+	m := controller.TemplateMapFromContext(ctx)
+	m["user"] = user
+	c.h.RenderHTML(w, "users/new", m)
```
Comment (sethvargo) [inline pkg/controller/user/update.go:109]: It's a little weird that `renderUpdate` is named "new"

File: pkg/controller/user/user.go [hunk 3]
```diff
@@ -31,24 +32,33 @@ import (
...
+	db         *database.Database
+	h          *render.Renderer
+	logger     *zap.SugaredLogger
 }
 
 // New creates a new controller for managing users.
-func New(ctx context.Context, client *auth.Client, cacher cache.Cacher, config *config.ServerConfig, db *database.Database, h *render.Renderer) *Controller {
+func New(
+	ctx context.Context,
+	fbInternal *iFB.Client,
```
Comment (sethvargo) [inline pkg/controller/user/user.go:47]: So I actually have this dream of moving everything to `internal/firebase`, but that's for another day 😄

**Review by sethvargo — APPROVED** [review]
/lgtm

File: pkg/controller/user/update.go (hunk 2 above)
Comment (whaught) [reply pkg/controller/user/update.go:109]: new and update share the same page...

**Pull request merged**